crates/iroha_core/src/smartcontracts/isi/nft.rs

```diff
@@ -0,0 +1,203 @@
+//! This module contains [`Nft`] instructions and queries implementations.
+
+use iroha_telemetry::metrics;
+
+use super::prelude::*;
+
+impl Registrable for NewNft {
+    type Target = Nft;
+
+    #[must_use]
+    #[inline]
+    fn build(self, authority: &AccountId) -> Self::Target {
+        Self::Target {
+            id: self.id,
+            content: self.content,
+            owned_by: authority.clone(),
+        }
+    }
+}
+
+/// ISI module contains all instructions related to NFTs:
+/// - register/unregister NFT
+/// - update metadata
+/// - transfer, etc.
+pub mod isi {
+    use iroha_data_model::{isi::error::RepetitionError, query::error::FindError};
+    use iroha_telemetry::metrics;
+
+    use super::*;
+
+    impl Execute for Register<Nft> {
+        #[metrics(+"register_nft")]
+        fn execute(
+            self,
+            authority: &AccountId,
+            state_transaction: &mut StateTransaction<'_, '_>,
+        ) -> Result<(), Error> {
+            let nft = self.object.build(authority);
+            let nft_id = nft.id.clone();
+
+            state_transaction
+                .world
+                .domain(&nft_id.domain)
+                .expect("INTERNAL BUG: Can't find domain of NFT to register");
```

Comment:
Shouldn't it be `FindError::Domain`?

Reply:
There are two options. If domain check is after NFT is found and removed, it should be `.expect("INTERNAL BUG ...")`. If before, it should be `FindError::Domain` (https://github.com/hyperledger-iroha/iroha/pull/5308#discussion_r1952094980). Changed to have domain check after.

Reply:
I see, that makes sense.